docs/how-to/topics/api-design.rst

```diff
@@ -0,0 +1,50 @@
+==========
+API design
+==========
+
+Toga's API is structured around the following principles:
+
+Properties
+==========
+
+Toga prefers to expose an object's state using property notation (e.g.
+``widget.property``) rather than getter or setter methods.
+
+Constructors
+============
+
+Any of a class's writable properties can be initialized in its constructor by passing
+keyword arguments with the same names. The constructor may also accept read-only
+properties such as :any:`Widget.id`, which cannot be changed later.
+
+If a constructor has a single required argument, such as the text of a :any:`Label`, it
+may be passed as a positional argument.
+
+Events
+======
+
+Events are used to notify your app of user actions. To make your app handle an event,
+you can assign either a regular or async callable to an event handler property. These
+can be identified by their names, which always begin with ``on_``.
```

Comment:
Also - event names (when they're not a common name) attempt to capture the "high level purpose", not the specific mode of interaction. For example:

* `on_press`, not `on_click` on buttons - "click" implies a mouse is used.
* `on_primary_action` and `on_secondary_action`, not "left click" and "right click"; again, the action might be a swipe, or a click, or a double click.